File: app/src/main/java/fr/xgouchet/chronorg/ui/presenters/JumpListPresenter.java
package fr.xgouchet.chronorg.ui.presenters;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.deezer.android.counsel.annotations.Trace;

import fr.xgouchet.chronorg.data.models.Entity;
import fr.xgouchet.chronorg.data.models.Jump;
import fr.xgouchet.chronorg.data.repositories.JumpRepository;
import fr.xgouchet.chronorg.ui.contracts.presenters.BaseListPresenter;
import rx.Observable;

/**
 * @author Xavier Gouchet
 */
@Trace
public class JumpListPresenter extends BaseListPresenter<Jump> {

    @NonNull private final JumpRepository jumpRepository;
    @Nullable private Entity entity;

    public JumpListPresenter(@NonNull JumpRepository jumpRepository) {
        this.jumpRepository = jumpRepository;
    }

    public void setEntity(@NonNull Entity entity) {
        this.entity = entity;
    }

    @Override protected Observable<Jump> getItemsObservable() {
        return jumpRepository.getJumpsInEntity(entity.getId());
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/presenters/JumpListPresenter.kt
package fr.xgouchet.khronorg.ui.presenters

import fr.xgouchet.khronorg.data.models.Jump
import fr.xgouchet.khronorg.data.models.Traveller
import fr.xgouchet.khronorg.data.query.QueryAlterer
import fr.xgouchet.khronorg.data.repositories.BaseRepository
import fr.xgouchet.khronorg.provider.KhronorgSchema.Companion.COL_ORDER
import fr.xgouchet.khronorg.provider.KhronorgSchema.Companion.COL_TRAVELLER_ID
import fr.xgouchet.khronorg.ui.navigators.JumpNavigator
import io.reactivex.Observable

/**
 * @author Xavier F. Gouchet
 */
class JumpListPresenter(val repository: BaseRepository<Jump>, val traveller: Traveller, navigator: JumpNavigator)
    : BaseListPresenter<Jump>(navigator) {


    override fun getItemsObservable(): Observable<Jump> {
        val alteration = QueryAlterer.where {
            equals(COL_TRAVELLER_ID, traveller.id.toString())
            orderAsc(COL_ORDER)
        }

        return repository.getWhere(alteration)
    }

    override fun itemSelected(item: Jump?) {
        if (item == null) {
            navigator.goToItemCreation()
        } else {
            navigator.goToItemDetails(item)
        }
    }

}